src/arraytraits.rs

```diff
@@ -94,6 +94,62 @@ impl<S, D, I> IndexMut<I> for ArrayBase<S, D>
     }
 }
 
+impl<'a, A, D> ArrayView<'a, A, D>
+where
+    D: Dimension,
+{
+    /// Get a reference of a element through the view.
+    ///
+    /// This is a replacement of `Index::index` since it forces us a wrong lifetime
+    /// https://github.com/bluss/rust-ndarray/issues/371
```

Comment:
I can edit this doc myself. We're going to use a matter-of-fact approach, but still be positive. This is a version of `index` that gives out references of a longer lifetime. :smile: